src/main/java/com/example/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.cart.items.CartItemDto;
+import com.example.bookstore.dto.cart.items.CreateCartItemRequestDto;
+import com.example.bookstore.dto.shopping.cart.ShoppingCartDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.CartItemMapper;
+import com.example.bookstore.mapper.ShoppingCartMapper;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.BookRepository;
+import com.example.bookstore.repository.CartItemRepository;
+import com.example.bookstore.repository.ShoppingCartRepository;
+import com.example.bookstore.security.CustomUserDetailsService;
+import com.example.bookstore.service.ShoppingCartService;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CustomUserDetailsService userDetailsService;
+
+    @Override
+    public ShoppingCartDto getShoppingCartByUserId(Authentication authentication) {
+        User user = (User) userDetailsService.loadUserByUsername(authentication.getName());
```

Comment:
why is cast necessary? shouldn't you service return `User`?

Reply:
I have to bring the basket back here, don't I?

Reply:
@teract10s you didn't answer the question :)

why casting to `User` is necessary here?

Reply:
@NazarSmith228 Because i have to get Shopping Cart by user